src/commands/google.ts

```diff
@@ -0,0 +1,98 @@
+import {CommandType} from "../types";
+import {logger, Config} from "@/config";
+
+import {
+  CacheType,
+  SlashCommandBuilder,
+  ChatInputCommandInteraction,
+  EmbedBuilder,
+  Colors,
+} from "discord.js";
+
+// Interfaces for getting specific entries.
+interface SearchResult {
+  title: string;
+  link: string;
+}
+
+interface Results {
+  items: SearchResult[];
+}
+
+const googleModule: CommandType = {
+  data: new SlashCommandBuilder()
+    .setName("google")
+    .setDescription("Google something and return the top 10 results.")
+    .addStringOption((option) =>
```

Comment:
As of right now, there is no max for the query length, the command fails if the query is too long. You can set a max by adding ``.setMaxLength(n)``